app-ms/services/ppt_to_md.py

```
@@ -0,0 +1,142 @@
+
+    with ZipFile(ppt_path) as archive:
+        slide_entries: list[tuple[int, str]] = []
+        for name in archive.namelist():
+            match = SLIDE_PATH_RE.match(name)
+            if match:
+                slide_entries.append((int(match.group(1)), name))
+        slide_entries.sort()
+
+        for number, slide_name in slide_entries:
+            xml_bytes = archive.read(slide_name)
+            root = ET.fromstring(xml_bytes)
```

Ruff (0.14.2)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)